.github/workflows/test.yml

```diff
@@ -1,5 +1,8 @@
 name: Run PyTest
 
+permissions:
```

Comment:
Could you tell me as to why we need this? You could link docs or something where you found out that this is needed.

Reply:
I was just trying to debug the permission issue, i saw that [here](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token)